packages/camera/camera/ios/Classes/FLTThreadSafeEventChannel.m

```diff
@@ -0,0 +1,35 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "FLTThreadSafeEventChannel.h"
+
+@interface FLTThreadSafeEventChannel ()
+@property(nonatomic, strong) FlutterEventChannel *channel;
+@end
+
+@implementation FLTThreadSafeEventChannel
+
+- (instancetype)initWithEventChannel:(FlutterEventChannel *)channel {
+  self = [super init];
+  if (self) {
+    _channel = channel;
+  }
+  return self;
+}
+
+- (void)setStreamHandler:(NSObject<FlutterStreamHandler> *)handler
+              completion:(void (^)(void))completion {
+  void (^block)(void) = ^{
+    [self.channel setStreamHandler:handler];
+    completion();
+  };
+
+  if (!NSThread.isMainThread) {
+    dispatch_async(dispatch_get_main_queue(), block);
```

Comment:
Oh I used `dispatch_sync` last time because I intended to just swap the call. But since I already read through the whole plugin code now and understand how it works, I switched it to `dispatch_async` as suggested last year. 

I still plan to use `dispatch_sync` as a replacement to the CAS spin-lock. So to keep the consistency I use GCD everywhere.